### DISCUSSION TITLE
進程和線程的差異與用途

### DISCUSSION THREAD

**Discussion by Shutaruku** [YuanData/LeetCode#3] [Q&A] [answered]
進程（Process）和線程（Thread）的差異與用途？

**Comment by YuanData** [comment, marked as answer]
| 特性         | 進程（Process）                                                                                   | 線程（Thread）                                                                             |
|----------|------------------------------------------------------------------|----------------------------------------|
| **定義**       | 操作系統動態執行的基本單位，是一個活動的程序執行實例。包含代碼、數據、堆棧等及操作系統資源。                              | 進程中的一個獨立執行流程，是CPU調度的最小單位。共享進程的資源，但有自己的執行堆棧和程序計數器。                 |
| **資源分配**    | 是資源分配的單位，擁有獨立的地址空間。                                                              | 不獨立擁有資源，與同一進程的其他線程共享資源和地址空間。                                                  |
| **獨立性**     | 進程間相互獨立，互不影響。                                                                      | 線程間相互依賴，一個線程的失敗可能影響到同一進程中的其他線程。                                           |
| **創建/銷毀開銷** | 相對較大，因為需要分配獨立的地址空間和其他資源。                                                        | 相對較小，因為共享部分資源，創建和銷毀的開銷更低。                                                      |
| **通信方式**    | 需要操作系統提供的特殊機制，如管道、消息隊列、共享內存等。                                               | 通信更為簡單，因為自然共享同一進程的記憶體空間和資源。                                                 |
| **用途**       | 提供了程序執行的獨立環境，適用於需要獨立運作的應用場景。                                                  | 提高應用程序的性能，適用於需要進行細粒度控制和高效率并行處理的場景。                                      |
